### PR TITLE
Fix google-payment issues

### PR DESCRIPTION
**Pull request by anton-karunarathne-aligent**
Fix google-payment issues

### CODE REVIEW

File: Model/Gateway.php [hunk 1]
```diff
@@ -129,31 +130,46 @@ public function __construct(
...
+     * @param $amount
+     * @param $reference
+     * @param $cvv
+     * @param $fraud_data
+     * @param $paymentType
      * @return mixed
      * @throws \Exception
      */
-    public function tokenPurchase($token, $amount, $reference, $cvv = null, $fraud_data = null)
+    public function tokenPurchase($pmntsToken, $amount, $reference, $cvv = null, $fraud_data = null, $paymentType=null)
```
Comment (dominicaligent) [inline Model/Gateway.php]: space in `$paymentType=null`

File: Model/Gateway.php [hunk 2]
```diff
@@ -129,31 +130,46 @@ public function __construct(
...
             $int_amount = (int)$multiplied;
         }
-        $payload = [
-            "customer_ip" => $customer_ip,
-            "card_token" => $token,
-            "cvv" => $cvv,
-            "amount" => $int_amount,
-            "reference" => $reference
+
+        if ($paymentType && $paymentType == self::PMNTS_GOOGLE_PAYMENT_METHOD_CODE) {
```
Comment (dominicaligent) [inline Model/Gateway.php]: Do we need `$paymentType &&`?

File: Model/Gateway.php [hunk 3]
```diff
@@ -129,31 +130,46 @@ public function __construct(
...
+            $token = json_decode($pmntsToken, true);
```
Comment (dominicaligent) [inline Model/Gateway.php:160]: Since this is a public function can you check after decode the values converting to the expected array?

**Review by dominicaligent — CHANGES_REQUESTED** [review]

**Review by anton-karunarathne-aligent — COMMENTED** [review]
@dominicaligent Mentioned issues are fixed. Please Review.

**Review by dominicaligent — APPROVED** [review]

**Pull request merged**